selfdrive/ui/layouts/main.py

```diff
@@ -34,23 +33,14 @@ def __init__(self):
     self._setup_callbacks()
 
   def _render(self, rect):
-    self._current_callback = None
-
     self._update_layout_rects(rect)
     self._handle_onroad_transition()
     self._render_main_content()
 
-    if self._current_callback:
-      self._current_callback()
-
   def _setup_callbacks(self):
-    self._sidebar.set_callbacks(
-      on_settings=lambda: setattr(self, '_current_callback', self._on_settings_clicked),
-      on_flag=lambda: setattr(self, '_current_callback', self._on_flag_clicked),
-    )
-    self._layouts[MainState.SETTINGS].set_callbacks(
-      on_close=lambda: setattr(self, '_current_callback', self._set_mode_for_state)
-    )
+    self._sidebar.set_callbacks(on_settings=self._on_settings_clicked,
+                                on_flag=self._on_flag_clicked)
+    self._layouts[MainState.SETTINGS].set_callbacks(on_close=self._set_mode_for_state)
     self._layouts[MainState.ONROAD].on_click = self._on_onrad_clicked
```

Comment:
@deanlee let's decide on we should use set functions or not soon, this is two different patterns for no reason

Reply:
think using a set method for is clearer than direct assignment

Reply:
What are your thoughts?

Reply:
https://github.com/commaai/openpilot/pull/35493